plugin/trino-base-jdbc/src/test/java/io/trino/plugin/jdbc/TestJdbcClient.java

```diff
@@ -109,4 +112,23 @@ public void testMetadataWithFloatAndDoubleCol()
                 new JdbcColumnHandle("COL3", JDBC_DOUBLE, DOUBLE),
                 new JdbcColumnHandle("COL4", JDBC_REAL, REAL)));
     }
+
+    @Test
+    public void testRenameTable()
+    {
+        String schemaName = "test_schema";
+        SchemaTableName oldTable = new SchemaTableName(schemaName, "foo");
+        SchemaTableName newTable = new SchemaTableName(schemaName, "bar");
+        ConnectorTableMetadata tableMetadata = new ConnectorTableMetadata(
+                oldTable,
+                ImmutableList.of(new ColumnMetadata("text", VARCHAR)));
+
+        jdbcClient.createSchema(session, schemaName);
+        jdbcClient.createTable(session, tableMetadata);
+        jdbcClient.renameTable(session, jdbcClient.getTableHandle(session, oldTable).get(), newTable);
+        jdbcClient.dropTable(session, jdbcClient.getTableHandle(session, newTable).get());
+        jdbcClient.dropSchema(session, schemaName);
+        assertThat(jdbcClient.getTableNames(session, Optional.empty())).doesNotContain(oldTable).doesNotContain(newTable);
+        assertThat(jdbcClient.getSchemaNames(session)).doesNotContain(schemaName);
+    }
```

Comment:
I added such tests to both classes. Since there was already a `testRenameTable` in `AbstractTestDistributedQueries()` I reused it a bit. Tests in `BaseSmokeConnectorTest` are a bit different since @kokosing asked to add INSERT and UPDATE statements there.